src/main/scala/io/iohk/ethereum/jsonrpc/EthService.scala

```diff
@@ -487,7 +487,8 @@ class EthService(
   def submitWork(req: SubmitWorkRequest): ServiceResponse[SubmitWorkResponse] = {
     reportActive()
     blockGenerator.getPrepared(req.powHeaderHash) match {
-      case Some(block) if appStateStorage.getBestBlockNumber() <= block.header.number =>
+      case Some(pendingBlock) if appStateStorage.getBestBlockNumber() <= pendingBlock.block.header.number =>
+        import pendingBlock._
         syncingController ! MinedBlock(block.copy(header = block.header.copy(nonce = req.nonce, mixHash = req.mixHash)))
         Future.successful(Right(SubmitWorkResponse(true)))
```

Comment:
Not related to your work, but I noticed this function accesses the DB so it should wrap the body in a `Future` rather than return a `Future.successful(...)`.